### ISSUE TITLE
wrong "overload is not iterable" error

### ISSUE THREAD

**Issue by geepy** [microsoft/pylance-release#27]
## Environment data
-   Language Server version: 2020.6.1
-   OS and version: Win10 1809
-   Python version (& distribution if applicable, e.g. Anaconda): Anaconda 3.1, Python 3.6.10

## Expected behaviour

code should not contain any errors

## Actual behaviour

pylance shows error "Overload[() -> (Module | NoseTester), () -> (Module | NoseTester)]" is not iterable 

## Logs

```
[Info  - 08:05:10] Pylance language server 2020.6.1 starting
[Info  - 08:05:10] Server root directory: c:\Users\XXX\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server
Python 3.6.10 :: Anaconda, Inc.
Python 3.6.10 :: Anaconda, Inc.
Python 3.6.10 :: Anaconda, Inc.
[Info  - 10:07:03] No configuration file found.
[Info  - 10:07:03] Setting pythonPath for service "Python": "C:\Users\XXX\AppData\Local\conda\envs\YYY\python.exe"
[Info  - 10:07:03] stubPath c:\My\Projects\Python\typings is not a valid directory.
Python 3.6.10 :: Anaconda, Inc.
[Info  - 10:07:04] Unable to get Python version from interpreter. Received ""
[Info  - 10:07:04] Assuming Python platform Windows
[Info  - 10:07:04] Searching for source files
[Info  - 10:07:04] Found 51 source files
[BG] analyzing: c:\My\Projects\Python\volcap\mailtest.py ...
[BG]   parsing: c:\My\Projects\Python\volcap\mailtest.py (99ms)
[BG]   parsing: c:\Users\XXX\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server\typeshed-fallback\stdlib\2and3\builtins.pyi (316ms)
[BG]   binding: c:\Users\XXX\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server\typeshed-fallback\stdlib\2and3\builtins.pyi (49ms)
[BG]   binding: c:\My\Projects\Python\volcap\mailtest.py (1ms)
[BG]   checking: c:\My\Projects\Python\volcap\mailtest.py ...
[BG]     parsing: c:\Users\XXX\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server\bundled-stubs\pandas\__init__.pyi (117ms)
[BG]     binding: c:\Users\XXX\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server\bundled-stubs\pandas\__init__.pyi (9ms)
[BG]     parsing: c:\Users\XXX\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server\bundled-stubs\pandas\core.pyi (237ms)
[BG]     binding: c:\Users\XXX\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server\bundled-stubs\pandas\core.pyi (65ms)
[BG]     parsing: c:\Users\XXX\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server\typeshed-fallback\stdlib\3\typing.pyi (31ms)
[BG]     binding: c:\Users\XXX\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server\typeshed-fallback\stdlib\3\typing.pyi (10ms)
[BG]     parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\__init__.py (279ms)
[BG]     binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\__init__.py ...
[BG]       parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\__init__.py (129ms)
[BG]       binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\__init__.py ...
[BG]         parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\numeric.py (191ms)
[BG]         binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\numeric.py ...
[BG]           parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\umath.py (27ms)
[BG]           binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\umath.py ...
[BG]             parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\_multiarray_umath.cp36-win_amd64.pyd (349ms)
[BG]             binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\_multiarray_umath.cp36-win_amd64.pyd (0ms)
[BG]           binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\umath.py (349ms)
[BG]           parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\numerictypes.py (60ms)
[BG]           binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\numerictypes.py (3ms)
[BG]           parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\fromnumeric.py (51ms)
[BG]           binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\fromnumeric.py (4ms)
[BG]           parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\arrayprint.py (90ms)
[BG]           binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\arrayprint.py (9ms)
[BG]           parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\_asarray.py (5ms)
[BG]           binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\_asarray.py (1ms)
[BG]           parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\_ufunc_config.py (20ms)
[BG]           binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\_ufunc_config.py (1ms)
[BG]         binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\numeric.py (627ms)
[BG]         parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\records.py (18ms)
[BG]         binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\records.py (5ms)
[BG]         parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\memmap.py (24ms)
[BG]         binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\memmap.py (8ms)
[BG]         parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\function_base.py (30ms)
[BG]         binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\function_base.py (2ms)
[BG]         parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\machar.py (49ms)
[BG]         binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\machar.py (2ms)
[BG]         parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\getlimits.py (21ms)
[BG]         binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\getlimits.py (4ms)
[BG]         parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\shape_base.py (20ms)
[BG]         binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\shape_base.py (2ms)
[BG]         parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\einsumfunc.py (41ms)
[BG]         binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\einsumfunc.py (6ms)
[BG]       binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\__init__.py (1059ms)
[BG]       parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\__init__.py (22ms)
[BG]       binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\__init__.py ...
[BG]         parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\type_check.py (7ms)
[BG]         binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\type_check.py (1ms)
[BG]         parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\index_tricks.py (64ms)
[BG]         binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\index_tricks.py (3ms)
[BG]         parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\function_base.py (168ms)
[BG]         binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\function_base.py (18ms)
[BG]         parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\nanfunctions.py (31ms)
[BG]         binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\nanfunctions.py (4ms)
[BG]         parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\shape_base.py (68ms)
[BG]         binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\shape_base.py (3ms)
[BG]         parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\stride_tricks.py (3ms)
[BG]         binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\stride_tricks.py (1ms)
[BG]         parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\twodim_base.py (6ms)
[BG]         binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\twodim_base.py (2ms)
[BG]         parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\ufunclike.py (5ms)
[BG]         binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\ufunclike.py (1ms)
[BG]         parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\histograms.py (45ms)
[BG]         binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\histograms.py (7ms)
[BG]         parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\polynomial.py (146ms)
[BG]         binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\polynomial.py (9ms)
[BG]         parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\utils.py (56ms)
[BG]         binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\utils.py (5ms)
[BG]         parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\arraysetops.py (6ms)
[BG]         binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\arraysetops.py (4ms)
[BG]         parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\npyio.py (158ms)
[BG]         binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\npyio.py (7ms)
[BG]         parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\financial.py (20ms)
[BG]         binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\financial.py (2ms)
[BG]         parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\arraypad.py (8ms)
[BG]         binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\arraypad.py (2ms)
[BG]         parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\_version.py (2ms)
[BG]         binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\_version.py (1ms)
[BG]       binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\lib\__init__.py (866ms)
[BG]       parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\matrixlib\__init__.py (2ms)
[BG]       binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\matrixlib\__init__.py ...
[BG]         parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\matrixlib\defmatrix.py (15ms)
[BG]         binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\matrixlib\defmatrix.py (3ms)
[BG]       binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\matrixlib\__init__.py (18ms)
[BG]     binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\__init__.py (2101ms)
[BG]     parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\multiarray.py (14ms)
[BG]     binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\core\multiarray.py (2ms)
[BG]     parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\testing\__init__.py (20ms)
[BG]     binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\testing\__init__.py ...
[BG]       parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\testing\_private\utils.py (199ms)
[BG]       binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\testing\_private\utils.py (8ms)
[BG]     binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\testing\__init__.py (209ms)
[BG]     parsing: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\testing\_private\nosetester.py (102ms)
[BG]     binding: C:\Users\XXX\AppData\Local\conda\envs\YYY\lib\site-packages\numpy\testing\_private\nosetester.py (2ms)
[BG]     parsing: c:\Users\XXX\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server\typeshed-fallback\stdlib\2and3\_typeshed\__init__.pyi (4ms)
[BG]     binding: c:\Users\XXX\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server\typeshed-fallback\stdlib\2and3\_typeshed\__init__.pyi (0ms)
[BG]   checking: c:\My\Projects\Python\volcap\mailtest.py (3251ms)
[BG] analyzing: c:\My\Projects\Python\volcap\mailtest.py (3716ms)
[BG] analyzing: c:\My\Projects\Python\_HiddenFile_c9362fe61e9245829d97d68608dc8bfa.py ...
[BG]   parsing: c:\My\Projects\Python\_HiddenFile_c9362fe61e9245829d97d68608dc8bfa.py (0ms)
[BG]   binding: c:\My\Projects\Python\_HiddenFile_c9362fe61e9245829d97d68608dc8bfa.py (0ms)
[BG]   checking: c:\My\Projects\Python\_HiddenFile_c9362fe61e9245829d97d68608dc8bfa.py (1ms)
[BG] analyzing: c:\My\Projects\Python\_HiddenFile_c9362fe61e9245829d97d68608dc8bfa.py (1ms)
[FG] parsing: c:\My\Projects\Python\volcap\mailtest.py (80ms)
[FG] parsing: c:\Users\XXX\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server\typeshed-fallback\stdlib\2and3\builtins.pyi (306ms)
[FG] binding: c:\Users\XXX\.vscode\extensions\ms-python.vscode-pylance-2020.6.1\server\typeshed-fallback\stdlib\2and3\builtins.pyi (49ms)
[FG] binding: c:\My\Projects\Python\volcap\mailtest.py (0ms)
[BG] analyzing: c:\My\Projects\Python\volcap\mailtest.py ...
[BG]   parsing: c:\My\Projects\Python\volcap\mailtest.py (0ms)
[BG]   binding: c:\My\Projects\Python\volcap\mailtest.py (0ms)
[BG]   checking: c:\My\Projects\Python\volcap\mailtest.py (30ms)
[BG] analyzing: c:\My\Projects\Python\volcap\mailtest.py (30ms)
```

## Code Snippet / Additional information

```python
import pandas as pd


df = pd.DataFrame([[1, 2, 3]], columns=['one', 'two', 'three'])
colnames = [x for x in df.columns.values]  # pylance error for 'df.columns.values': "Overload[() -> (Module | NoseTester), () -> (Module | NoseTester)]" is not iterable
print(colnames)

```

**Comment by erictraut** [comment]
@gramster, this looks like a bug in the pandas type stubs. The `values` methods are not marked with @property. Also, it looks like you were trying to use an @overload decorator for the `values` property getter, but @overload works only with functions, and a property is an object, so you unfortunately can't overload a property.

**Comment by gramster** [comment]
Should be fixed in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.8.1, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#202081-13-august-2020
